AutomatedLabWorker/AutomatedLabWorkerVirtualMachines.psm1

```diff
@@ -425,6 +425,55 @@ function New-LWHypervVM
         $referenceDiskPath = if ($Machine.ReferenceDiskPath) { $Machine.ReferenceDiskPath } else { $Machine.OperatingSystem.BaseDiskPath }
         $systemDisk = New-VHD -Path $path -Differencing -ParentPath $referenceDiskPath -ErrorAction Stop
         Write-PSFMessage "`tcreated differencing disk '$($systemDisk.Path)' pointing to '$ReferenceVhdxPath'"
+
+        if ($Machine.InitialDscLcmConfigurationMofPath -or $Machine.InitialDscConfigurationMofPath)
+        {
+            $mountedOsDisk = Mount-VHD -Path $path -Passthru
```

Comment:
Is it correct that this feature only works for Hyper-V hosted machines and not for Azure?

Reply:
Yes, unfortunately we have no simple way of doing that. The DSC extension (which would need to be used) requires the configuration script and all dependent modules packaged in an archive that is uploaded to e.g. a storage account. The MOF cannot be applied at all. On Azure, the scenario that machines have no accessible networking is rather uncommon though. I mainly enabled this because my clients deploy on Hyper-V clusters and don't have connectivity from their host clusters to the VMs. We still wanted to be able to configure whatever we needed, and this was an easy solution

Reply:
Perfect, thanks!